unpack

```diff
@@ -2,70 +2,38 @@
 # unpack - unarchive archive files
 # this file in public domain
 
-usage()
+die()
 {
-	echo "usage: unpack file [destdir]" >&2
+	printf "%s\n" "$*" >&2
 	exit 1
 }
 
+crun()
+{
+	command -v "$1" 2>&1 >/dev/null ||
+		die "unpack: utility $1 is not installed"
+
+	"$@" || die "failed to extract"
+}
+
 case "$#" in
-(1) set -- "$1" "$PWD" ;;
-(2)                    ;;
-(*) usage              ;;
+	(1) set -- "$1" "$PWD" ;;
+	(2) ;;
+	(*) die "usage: unpack file [destdir]" ;;
 esac
 
-if [ ! -f "$1" ]
-then
-	printf "unpack: %s: not a regular file\n" "$1"
-	exit 1
-fi
-
-if [ ! -d "$2" ]
-then
-	printf "unpack: %s: not a regular file\n" "$2"
-	exit 1
-fi
+! [ -e "$1" ] && die "unpack: $1: not a valid file"
```

Comment:
but do the same thing the latter is just less readable.

Reply:
that's the point; im not willing to rewrite or make things unwork.